torchrl/envs/libs/pettingzoo.py

```diff
@@ -0,0 +1,929 @@
+import copy
+from typing import Dict, List, Optional, Tuple, Union
+
+import torch
+from tensordict.tensordict import TensorDictBase
+
+from torchrl.data import (
+    CompositeSpec,
+    DiscreteTensorSpec,
+    OneHotDiscreteTensorSpec,
+    UnboundedContinuousTensorSpec,
+)
+from torchrl.envs.common import _EnvWrapper
+from torchrl.envs.libs.gym import _gym_to_torchrl_spec_transform, set_gym_backend
+from torchrl.envs.utils import _check_marl_grouping, _replace_last, MarlGroupMapType
+
+IMPORT_ERR = None
+try:
+    import pettingzoo
+
+    _has_pettingzoo = True
+
+except ImportError as err:
+    _has_pettingzoo = False
+    IMPORT_ERR = err
+
```

Comment:
We should use local imports (if possible) and do `_has_pettingzoo = importlib.util.find_spec("pettingzoo") is not None` to reduce the time it takes to load the lib

Reply:
what do you mean local imports? 
i have updated to the spec finding code proposed